S3/S3.py

```diff
@@ -317,19 +317,148 @@ def list_all_buckets(self):
         response["list"] = getListFromXml(response["data"], "Bucket")
         return response
 
-    def bucket_list(self, bucket, prefix = None, recursive = None, uri_params = None, limit = -1):
+    def bucket_list(
+        self,
+        bucket,
+        prefix=None,
+        recursive=None,
+        uri_params=None,
+        limit=-1,
+        list_objects_v2=False
+    ):
+        if uri_params is None:
+            uri_params = {}
+        if uri_params.get("list_type") == "v2":
+            list_objects_v2 = True
+
         item_list = []
         prefixes = []
-        for truncated, dirs, objects in self.bucket_list_streaming(bucket, prefix, recursive, uri_params, limit):
-            item_list.extend(objects)
-            prefixes.extend(dirs)
+        if list_objects_v2:
+            uri_params.update({"list_type": "v2"})
+            for truncated, dirs, objects in self.bucket_list_streaming(
```

Comment:
isn't this supposed to be bucket_list_v2_streaming?

Reply:
Fixed. Thanks